File: langs/C/pointer/change.cpp
//https://www.cnblogs.com/heat-man/p/4646051.html
//pointer
#include<stdio.h>
void changeSucceed(int *   &pp)
{
        int b=7;
        printf("Change pp=%p &pp=%p\n", pp, &pp);
        pp = &b;
}
int main()
{
        int *p=NULL;
        int a=5;
        p = &a;
        printf("main p=%p &p=%p\n", p, &p);
        changeSucceed(p);
        printf("%d\n",*p);   //5
        getchar();
        return 0;
}

File: langs/C/pointer/change.js
function changeFail(pp){
	var b = 7;
	pp = b;
}
function change(pp){
	var b = 7;
	return pp = b;
}
var a = 5;
// changeFail(a);
a = change(a)    //如果不赋值的话，又有什么办法改变a的值呢？这种指针的问题在JS中没有应用吧。

console.log(a)







